### PR TITLE
Fix opengl scene selection

### LINKED ISSUE #1750: Scene selection does not work for the OpenGL renderer
## Description of bug / unexpected behavior
Scene selection seemingly ignores the choice of Scene the user makes when using the OpenGL renderer. For example,

```py
from manim import *

class SphereTest(Scene):
    def construct(self):
        a = Sphere()
        self.play(Create(a))
        self.interactive_embed()

class P(Scene):
    def func(self, u, v):
        return np.array([np.cos(u) * np.cos(v), np.cos(u) * np.sin(v), u])

    def construct(self):
        axes = ThreeDAxes(x_range=[-4,4], x_length=8)
        surface = ParametricSurface(
            lambda u, v: axes.c2p(*self.func(u, v)),
            u_min=-PI,
            u_max=PI,
            v_min=0,
            v_max=TAU,
        )
        # self.set_camera_orientation(theta=70 * DEGREES, phi=75 * DEGREES)
        self.add(axes, surface)
        self.interactive_embed()
```

does correctly bring up the scene selector:

```
C:\Users\Daryl\manim-videos\manim-videos>manim -p --renderer=opengl opengltests.py
Manim Community v0.8.0

1: P
2: SphereTest

Choose number corresponding to desired scene/arguments.
(Use comma separated list for multiple entries)
Choice(s):  2
```

but no matter what scene I choose, it always renders the first scene.

## How to reproduce the issue

<details><summary>Code for reproducing the problem</summary>

```py
Paste your code here.
```

</details>

## Additional media files

<details><summary>Images/GIFs</summary>

</details>

## Logs
<details><summary>Terminal output</summary>

```
PASTE HERE OR PROVIDE LINK TO https://pastebin.com/ OR SIMILAR
```

</details>

## System specifications

<details><summary>System Details</summary>

- OS (with version, e.g Windows 10 v2004 or macOS 10.15 (Catalina)):
- RAM:
- Python version (`python/py/python3 --version`):
- Installed modules (provide output from `pip list`):
```
PASTE HERE
```
</details>

<details><summary>LaTeX details</summary>

+ LaTeX distribution (e.g. TeX Live 2020):
+ Installed LaTeX packages:

</details>

<details><summary>FFMPEG</summary>

Output of `ffmpeg -version`:

```
PASTE HERE
```
</details>

### PR DESCRIPTION
**Pull request by ryanmccauley211**
## Changelog / Overview

Fixed issue where scene selection was ignored when using the opengl renderer

## Motivation

Fixes #1750

Scene selection by the user was being overwritten when using the opengl renderer
## Explanation for Changes

Allows the user to select from multiple scenes when using the opengl renderer

[Documentation Reference](https://manimce--####.org.readthedocs.build)

## Testing Status

Tested locally with the code in the tracked issue

Update - The PR #1588 added a line to update `config["write_all"]` when the user was prompted for a scene. This caused no issue at the time as it only ran once, now with opengl we can have interactive mode and rerun scenes so this causes issues where it reruns scenes that are not selected. I added a separate flag to resolve this, not sure of another solution for this.

Some notes on behaviour as there are a lot of ways to run manim that all impact this logic in some ways making it a bit awkward to get right:

- When using the gui you can select any scene in the python file even if you only selected one to begin with, this behaviour seems fine to me for the gui
- When using interactive mode, rerunning a scene when only one is selected reruns that scene
- When using interactive mode, rerunning a scene when multiple scenes are selected cycles through the selected scenes

## Checklist
- [X] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [X] I have written a descriptive PR title (see top of PR template for examples)
- [X] I have written a changelog entry for the PR or deem it unnecessary
- [ ] My new functions/classes either have a docstring or are private
- [ ] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [ ] The PR title is descriptive enough
- [ ] The PR is labeled correctly
- [ ] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

File: manim/cli/render/commands.py [hunk 1]
```diff
@@ -103,8 +103,6 @@ def __repr__(self):
             try:
                 renderer = OpenGLRenderer()
                 while True:
-                    scene_classes = scene_classes_from_file(file)
```
Comment (eulertour) [inline manim/cli/render/commands.py:106]: By always rerunning the same scene this breaks the ability to dynamically choose new scenes from the GUI. https://github.com/ManimCommunity/manim/blob/723e645abce9c86cdf40c98959157e6ff93896a0/manim/gui/gui.py#L29

**Review by eulertour — CHANGES_REQUESTED** [review]

File: manim/cli/render/commands.py (hunk 1 above)
Comment (ryanmccauley211) [reply manim/cli/render/commands.py:106]: I updated it and tested with the GUI and it allows switching between scenes now

**Review by behackl — CHANGES_REQUESTED** [review]
This needs more work:

- `manim --renderer=opengl example.py` brought up the scene selection, but still always starts to render the first scene.
- And I mean "starts to render" quite literally: rendering doesn't end, at all.

**Comment by eulertour** [comment]
I was able to choose scenes from the GUI but trying to quit caused an infinite loop where the scene kept running over and over.

**Comment by ryanmccauley211** [comment]
> I was able to choose scenes from the GUI but trying to quit caused an infinite loop where the scene kept running over and over.

This loop issue has been fixed

**Comment by ryanmccauley211** [comment]
> This needs more work:
> 
> * `manim --renderer=opengl example.py` brought up the scene selection, but still always starts to render the first scene.
> * And I mean "starts to render" quite literally: rendering doesn't end, at all.

Fixed this

**Review by behackl — APPROVED** [review]
Works now! Thanks for your efforts!

**Pull request merged**